File: Day6-TodoList/app/src/main/java/com/byted/camp/todolist/ViewModel/TodoListViewModel.java
package com.byted.camp.todolist;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.byted.camp.todolist.Repository.TodoRepository;
import com.byted.camp.todolist.db.room.TodoEntity;

import java.util.List;

import io.reactivex.Completable;

public class TodoListViewModel extends AndroidViewModel {
    // Never pass context into ViewModel instances.
    // Do not store Activity, Fragment, or View instances or their Context in the ViewModel.
    private TodoRepository mRepository;


    private LiveData<List<TodoEntity>> allTodos;

    public TodoListViewModel(@NonNull Application application) {
        super(application);
        mRepository = new TodoRepository(application);
        allTodos = mRepository.getAllTodos();
    }

    public LiveData<List<TodoEntity>> getAllTodos() {
        return allTodos;
    }

    public Completable insert(TodoEntity todo) {
        return mRepository.insert(todo);
    }

    public Completable update(TodoEntity todo) {
        return mRepository.update(todo);
    }

    public Completable delete(TodoEntity todo) {
        return mRepository.delete(todo);
    }
}

File: Day6-TodoList/app/src/main/java/com/byted/camp/todolist/viewModel/TodoListViewModel.kt
package com.byted.camp.todolist.ViewModel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData

import com.byted.camp.todolist.Repository.TodoRepository
import com.byted.camp.todolist.db.room.TodoEntity

import io.reactivex.Completable

class TodoListViewModel(application: Application) : AndroidViewModel(application) {
    // Never pass context into ViewModel instances.
    // Do not store Activity, Fragment, or View instances or their Context in the ViewModel.
    private val mRepository: TodoRepository


    val allTodos: LiveData<List<TodoEntity>>

    init {
        mRepository = TodoRepository(application)
        allTodos = mRepository.allTodos
    }

    fun insert(todo: TodoEntity): Completable {
        return mRepository.insert(todo)
    }

    fun update(todo: TodoEntity): Completable {
        return mRepository.update(todo)
    }

    fun delete(todo: TodoEntity): Completable {
        return mRepository.delete(todo)
    }
}
